proposals/repo-to-tar/README.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Package Repository to Tar"
+authors: [ "Ashish Kumar <ashishndiitr@gmail.com>" ]
+status: "draft"
+approvers: [ ]
+---
+
+# <Package Repository to Tar>
+
+## Problem Statement
+
+Presently imgpkg does not support any way of creating and storing bundles in any format directly to local disk for sharing purposes, so as to remove the need of a registry in between. This proposal aims to add this functionality to imgpkg.
+
+Consider this workflow, where a user wants to create a tar file of a bundle image and then push it to a registry. <br>
+In order to obtain a tar file, one needs to first push the bundle image to the registry and then leverage the [command](https://carvel.dev/imgpkg/docs/v0.37.x/air-gapped-workflow/#option-2-with-intermediate-tarball) for copying as tar in the air gapped workflow.
+
+For example, In order to obtain a tar of bundle image of examples/basic-step-2, we need run the following commands in succession :<br>
+
+`imgpkg push -b index.docker.io/user1/simple-app-bundle:v1.0.0 -f examples/basic-step-2` <br>
+`imgpkg copy -b index.docker.io/user1/simple-app-bundle:v1.0.0 --to-tar /tmp/my-image.tar`
+
+This has 2 fold issues : <br>
+1. Dependent on using a registry to create a tar file from a bundle image.
+2. The tar file is created from the bundle image, which is not required if the user just wants to create a tar file having a layer as tar with configurations/metadata.
+
+## Terminology / Concepts
+1. Definitions of terms used in the proposal with respect to imgpkg can be found [here.](https://carvel.dev/imgpkg/docs/v0.37.x/)<br>
+2. Resources regarding docker save and docker load can be found [here.](https://docs.docker.com/engine/reference/commandline/save/)<br>
+3. A good article to understand OCI Artifacts can be found [here.](https://dlorenc.medium.com/oci-artifacts-explained-8f4a77945c13)
+
+## Proposal
+
+#### imgpkg
+imgpkg save command will be added to imgpkg. This command will create a tar file from the bundle image. The command will be used as follows : <br>
+`imgpkg save -f examples/basic-step-2 --to-tar /tmp/my-image.tar`<br>
```

Comment:
I like the proposed UX in this comment https://github.com/carvel-dev/imgpkg/issues/55#issuecomment-962209740. It is more in line with the rest of the commands that we already have in imgpkg.

For this proposal, we could also mention the idea of "inflating" the bundle with all the associated images, nevertheless, I do not think this is a priority to be implemented, but it would be good to have the proposal for it approved.

Reply:
I've looked at the comment and also modified the proposal based on our slack conversation.
I've added the inflate option in the future goals as well explaining it briefly.